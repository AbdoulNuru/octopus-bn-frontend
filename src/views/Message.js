import React from 'react';
import styles from '../styles/signup.module.css';
import Svg from '../components/SvgMap';
import translate from '../languages/translate';
import LanguageButtons from '../components/LanguageButtons';
import { withRouter } from 'react-router-dom';
import { BrowserRouter, Link, Redirect } from 'react-router-dom';

const Message = (props) => {
	const state = props.location.state;
	return (
		<div className={styles.container}>
			<div className={styles.side}>
				<div className={styles.text_wrap}>
					<h1>Barefoot Nomad</h1>
					<p>{translate('bn-value')}</p>
				</div>
			</div>
			<Svg />
			<div className={styles.forms}>
				<LanguageButtons />
				<div className={styles.text_wrap}>
					<div className={styles.fs_confirm} style={{ marginTop: '360px', fontSize: '1.1em' }}>
						{state.message.length != 0 && (
							<div>
								{translate(state.message)},{' '}
								{state.next && (
									<Link className={styles.linkLabel} to={state.next}>
										{translate(state.label)}
									</Link>
								)}
							</div>
						)}
						{}
					</div>
				</div>
			</div>
		</div>
	);
};

export default withRouter(Message);
