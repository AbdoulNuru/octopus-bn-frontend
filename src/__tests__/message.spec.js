import { Route, MemoryRouter } from 'react-router-dom';
import Message from '../views/Message';
import { render, fireEvent, cleanup, waitFor, waitForElement } from '@testing-library/react';
import React from 'react';

const renderWithRouter = (Message) =>
	render(
		<MemoryRouter state={{ message: 'password_reset', next: '/', label: 'sign-in-click' }}>
			<Route path="/message">{Message}</Route>
		</MemoryRouter>
	);

describe('RESET', () => {
	afterEach(cleanup);
	it('should render reset password component', () => {
		const { asFragment } = renderWithRouter();
		expect(asFragment(<Message />)).toMatchSnapshot();
	});
});
