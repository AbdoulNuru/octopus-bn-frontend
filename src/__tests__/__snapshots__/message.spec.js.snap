// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RESET should render reset password component 1`] = `<DocumentFragment />`;
